src/cbuild/hooks/post_install/007_strip_debug.py

```diff
@@ -159,5 +161,59 @@ def invoke(pkg):
     else:
         (pkg.destdir / "usr/lib").rmdir()
 
+    # now collect the referenced debug sources
+    for v in ddest.rglob("*"):
+        if v.is_dir():
+            continue
+
+        for source_file in (
+            chroot.enter(
+                "llvm-dwarfdump",
+                "--show-sources",
+                f"/{v.relative_to(pkg.rparent.bldroot_path)}",
+                capture_output=True,
+                check=True,
+                ro_root=True,
+                ro_build=True,
+                unshare_all=True,
+            )
+            .stdout.strip()
+            .splitlines()
+        ):
+            source_file = pathlib.Path(source_file.strip().decode())
+
+            # don't have access to these!
+            if source_file.name in ["<stdin>"]:
+                continue
+
+            # /usr sourcefiles like /usr/include headers are already provided by -devel packages
+            if source_file.is_relative_to("/usr"):
+                continue
+
+            if not source_file.is_absolute():
+                src = pkg.rparent.srcdir / source_file
+
+                if not src.exists():
+                    # silence warning for crt objects
+                    if source_file.name not in [
+                        "Scrt1.c",
+                        "crt1.c",
+                        "crtbegin.c",
+                    ]:
+                        pkg.log_warn(
+                            f"missing debug source file: {source_file}"
+                        )
+                else:
+                    dst = (
+                        ddest
+                        / f"usr/src/debug/{pkg.pkgname}-{pkg.pkgver}-r{pkg.pkgrel}"
+                        / source_file
+                    )
+                    dst.parent.mkdir(parents=True, exist_ok=True)
+                    shutil.copy(src, dst)
+            else:
+                pkg.log_warn(f"unknown debug source file: {source_file}")
+                continue
```

Comment:
```suggestion
            if source_file.is_absolute():
                pkg.log_warn(f"unknown debug source file: {source_file}")
                continue

            src = pkg.rparent.srcdir / source_file

            if not src.exists():
                # silence warning for crt object
                if source_file.name not in [
                    "Scrt1.c",
                    "crt1.c",
                    "crtbegin.c",
                ]:
                    pkg.log_warn(
                        f"missing debug source file: {source_file}"
                    )
                continue
             
            dst = (
                ddest
                / f"usr/src/debug/{pkg.pkgname}-{pkg.pkgver}-r{pkg.pkgrel}"
                / source_file
            )
            dst.parent.mkdir(parents=True, exist_ok=True)
            shutil.copy(src, dst)
```

el flatto (easier to read this way)